polyhost/core/poly_core.py

```
@@ -0,0 +1,515 @@
+        except Exception as e:  # never let a save attempt break shutdown/sleep
+            self.log.debug("MRU save request failed: %s: %s", type(e).__name__, e)
+
+    # ------------------------------------------------------------------
+    # Overlay mapping / active-window handler
+    # ------------------------------------------------------------------
+
+    def load_overlay_mapping(self, path):
+        import yaml
+        try:
+            with open(path, encoding="utf-8") as f:
+                self.mapping = yaml.load(f, Loader=yaml.FullLoader)
```

Ruff (0.15.15)

[error] Probable use of unsafe loader `FullLoader` with `yaml.load`. Allows instantiation of arbitrary objects. Consider `yaml.safe_load`.

(S506)